cmd/agent/flow_run.go

```diff
@@ -25,6 +27,24 @@ import (
 	_ "github.com/grafana/agent/component/all"
 )
 
+var (
+	configHash = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "agent_flow_config_hash",
+			Help: "Hash of the currently active config file.",
+		},
+		[]string{"sha256"},
+	)
+	configReloadSuccess = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "agent_flow_config_last_reload_successful",
```

Comment:
still not super happy about this one, as both flow and non-flow metrics show up in the metrics endpoint, instead of being registered lazily. Ideas welcome

Reply:
Do you mean that Flow metrics would show up even in the 'static' Agent mode?

If so, it might be a little annoying but not too much. We _could_ stuff everything into the Run method, but I'm not sure the readability is a good tradeoff.

Reply:
For Flow side: In `flow_run.go` instead of using promauto you could register it with the DefaultRegister. Same with static agent , instead of promauto register with the reg in `entrypoint.go`